mxnet-bot/EmailBot/EmailBot.py

```diff
@@ -0,0 +1,324 @@
+        self.sender = sender
+        self.recipients = [s.strip() for s in recipients.split(",")] if recipients else None
+        self.aws_region = aws_region
+        self.opendata = None
+        self.closeddata = None
+        self.start = datetime.datetime.strptime("2015-01-01", "%Y-%m-%d")
+        self.end = datetime.datetime.today()+datetime.timedelta(days=2)
+        self.sla = 5
+        # 2018-5-15 is the date that 'sla' concept was used.
+        self.sla_start = datetime.datetime.strptime("2018-05-15", "%Y-%m-%d")
+
+    def __clean_string(self, raw_string, sub_string):
+        """
+        This method is to convert all non-alphanumeric characters from raw_string to sub_string
+        """
+        cleans = re.sub("[^0-9a-zA-Z]", sub_string, raw_string)
+        return cleans.lower()
+
+    def __set_period(self, period):
+        """
+        This method is to set the time period. ie: set_period(7)
+        Because GitHub use UTC time, so we set self.end 2 days after today's date
+        For example:
+        self.today = "2018-07-10 00:00:00"
+        self.end = "2018-07-12 00:00:00"
+        self.start = "2018-07-04 00:00:00"
+        """
+        today = datetime.datetime.strptime(str(datetime.datetime.today().date()), "%Y-%m-%d")
+        self.end = today + datetime.timedelta(days=2)
+        timedelta = datetime.timedelta(days=period)
+        self.start = self.end - timedelta
+
+    def __count_pages(self, obj, state='all'):
+        """
+        This method is to count how many pages of issues/labels in total
+        obj could be "issues"/"labels"
+        state could be "open"/"closed"/"all", available to issues
+        """
+        assert obj in set(["issues", "labels"]), "Invalid Input!"
+        url = 'https://api.github.com/repos/{}/{}'.format(self.repo, obj)
+        if obj == 'issues':
+            response = requests.get(url, {'state': state},
+                                    auth=self.auth)
+        else:
+            response = requests.get(url, auth=self.auth)
+        assert response.status_code == 200, response.status_code
+        if "link" not in response.headers:
+            # That means only 1 page exits
+            return 1
+        # response.headers['link'] will looks like:
+        # <https://api.github.com/repositories/34864402/issues?state=all&page=387>; rel="last"
+        # In this case we need to extrac '387' as the count of pages
+        return int(self.__clean_string(response.headers['link'], " ").split()[-3])
+
+    def read_repo(self, periodically=True):
+        """
+        This method is to read issues in the repo.
+        if periodically == True, it will read issues which are created in a specific time period
+        if periodically == False, it will read all issues
+        """
+        logging.info("Start reading {} issues".format("periodically" if periodically else "all"))
+        if periodically:
+            self.__set_period(8)
```

Comment:
What does 8 means here?

Reply:
Well, it took me quiet a long time to find the correct period value... Github uses UTC time zone but we are in pacific time zone. '8' is the right value to set the week period without losing any issues.

Reply:
Could you please add comments in here?